nucleus/dataset.py

```diff
@@ -831,6 +836,51 @@ def create_slice(
         )
         return Slice(response[SLICE_ID_KEY], self._client)
 
+    def build_slice(
+        self,
+        name: str,
+        sample_size: int,
+        sample_method: Union[str, SliceBuilderMethods],
+        filters: Optional[SliceBuilderFilters] = None,
+    ) -> Union[str, AsyncJob]:
+        """Build a slice using Nucleus' Smart Sample tool. Allowing slices to be built
+        based on certain criteria, and filters.
+
+        Args:
+            name: Name for the slice being created. Must be unique per dataset.
+            sample_size: Size of the slice to create. Capped by the size of the dataset and the applied filters.
+            sample_method: How to sample the dataset, currently supports 'Random' and 'Uniqueness'
+            filters: Apply filters based on an existing slice, or autotag.
+
+        Examples:
+            from nucleus.slice import SliceBuilderFilters, SliceBuilderMethods, SliceBuilderFilterAutotag
+
+            # random slice
+            job = dataset.build_slice("RandomSlice", 20, SliceBuilderMethods.RANDOM)
+
+            # slice with filters
+            filters = SliceBuilderFilters(
+                slice_id="<some slice id>",
+                autotag=SliceBuilderFilterAutotag("tag_cd41jhjdqyti07h8m1n1", [-0.5, 0.5])
+            )
+            job = dataset.build_slice("NewSlice", 20, SliceBuilderMethods.RANDOM, filters)
+
+        Returns: An async job
+
+        """
+        payload = create_slice_builder_payload(
```

Comment:
You should probably validate `sample_method` that it is indeed a `SliceBuilderMethods`.